server/rest/src/main/java/org/infinispan/rest/resources/ServerResource.java

```diff
@@ -466,6 +474,33 @@ private CompletionStage<RestResponse> dataSourceTest(RestRequest restRequest) {
       }, invocationHelper.getExecutor());
    }
 
+   private CompletionStage<RestResponse> getCacheConfigDefaultAttributes(RestRequest request) {
+      Configuration configuration = new ConfigurationBuilder().build();
+      Map<String, Map<String, String>> attributes = new LinkedHashMap<>();
+      allAttributes(configuration, attributes, null);
+      return asJsonResponseFuture(Json.make(attributes), isPretty(request));
+   }
+
+   private static void allAttributes(ConfigurationElement<?> element, Map<String, Map<String, String>> attributes, String prefix) {
+      prefix = prefix == null ? "" : element.elementName();
+      Map<String, String> attributeMap = new LinkedHashMap<>();
+      for (Attribute<?> attribute : element.attributes().attributes()) {
+         AttributeDefinition<?> definition = attribute.getAttributeDefinition();
+         String value;
+         if (attribute.getInitialValue() == null) {
+            value = null;
+         } else {
+            value = attribute.getInitialValue().toString();
+         }
+         attributeMap.put(definition.name(), value);
+      }
+      attributes.put(prefix, attributeMap);
```

Comment:
Should we return something representing how it is nested in the configuration? IIUC, right now we have everything directly under the JSON root.

Reply:
@jabolina can you elaborate more, what and where exactly you want to represent?

Reply:
Maybe what I am saying is unnecessary. For example, key and value encodings are nested under the encoding configuration:

```java
new ConfigurationBuilder()
    .encoding()
        .key().mediaType(...)
    .encoding()
        .value().mediaType(...)
    .build();
```

But on the JSON, we have that:

```json
{
   "encoding":{
      "media-type": null
   },
   "key":{
      "media-type": null
   },
   "value":{
      "media-type": null
   }
}
```

Everything is under the root, it doesn't nest in the same way when configuring.

Reply:
The question for me is can I copy/paste these defaults and put it in to create a cache. This would ensure the layout is the same and that we know a config maps to the one it says it is.

Reply:
> But on the JSON, we have that:

@jabolina This nesting pattern for encoding we get is from `ConfigurationBuilder().build();`

Reply:
> The question for me is can I copy/paste these defaults and put it in to create a cache. This would ensure the layout is the same and that we know a config maps to the one it says it is.

I don't think this JSON of default values can be considered for creating a cache. For a cache configuration we need to have cache mode at the start of configuration, which can't be possible in this case.

Reply:
The defaults are literally what is created when the cache is just defined and not overridden though, so it better be able to define a cache with.